### PR TITLE
Chore/contact button amendments

### PR DESCRIPTION
**Pull request by Sepehr-Sobhani**
Zenhub: [695](https://github.com/bcgov/cas-cif/issues/695)

### CODE REVIEW

File: app/components/Form/NewContactButton.tsx [hunk 1]
```diff
@@ -0,0 +1,80 @@
...
+  connectionString,
+  projectContactFormId,
+  projectId,
+  contactIndex,
+}) => {
+  const router = useRouter();
+  const [addContact, isAddingContact] = useCreateNewContactFormChange();
+  const [addContactMutation] = useAddContactToRevision();
+
+  const clickHandler = (e: React.ChangeEvent<{}>) => {
```
Comment (matthieu-foucault) [inline app/components/Form/NewContactButton.tsx:29]: Should the project_contact form_change be created now, or when the user submits the contact form?

File: app/components/Contact/ContactForm.tsx [hunk 2]
```diff
@@ -3,18 +3,72 @@ import FormBase from "components/Form/FormBase";
...
     <FormBase
       {...props}
       schema={contactSchema as JSONSchema7}
       uiSchema={uiSchema}
+      // If we have contactIndex(this can be revisionId or connectionString as well) in the route query, we need to override the default onSubmit handler
+      // This is intended to be used for adding new contact and linking it to the project contact form
+      onSubmit={router?.query?.contactIndex ? handleSubmit : props.onSubmit}
     >
       <Button type="submit" style={{ marginRight: "1rem" }}>
         Submit
```
Comment (matthieu-foucault) [inline app/components/Contact/ContactForm.tsx:74]: the "discard" button below should redirect back to the project revision page

**Comment by matthieu-foucault** [comment]
@Sepehr-Sobhani there's some conflicts that need to be resolved, can you rebase this?

**Comment by Sepehr-Sobhani** [comment]
> @Sepehr-Sobhani there's some conflicts that need to be resolved, can you rebase this?

@matthieu-foucault On it

**Review by matthieu-foucault — CHANGES_REQUESTED** [review]
To add the button to the search dropdown widget, that widget could support a custom `uiSchema` option, e.g. `getAutocompleteProps: (WidgetProps) => AutocompletProps`, allowing you to customize the `Autocomplete` `ListboxComponent` prop.

File: app/components/Form/NewContactButton.tsx [hunk 3]
```diff
@@ -0,0 +1,61 @@
...
+
+interface Props {
+  projectRevisionRowId: number;
+  connectionString: string;
+  projectContactFormId: string;
+  projectId: number;
+  contactIndex: number;
+}
+
+const NewContactButton: React.FC<Props> = ({
```
[the remark this answers is not in the document]
Comment (Sepehr-Sobhani) [reply app/components/Form/NewContactButton.tsx:17]: @matthieu-foucault Based on [this updated comment](https://github.com/bcgov/cas-cif/issues/695#issuecomment-1149232780), we needed to change its location.

[the remark this answers is not in the document]
Comment (matthieu-foucault) [reply app/components/Form/NewContactButton.tsx:17]: In that case, I would argue that the button should use the `secondary` variant. We should only have one primary per page.

**Review by matthieu-foucault — DISMISSED** [review]
There are some margin issues that we need to fix (too much space between a search dropdown and the button), but I think they need to be done at the field template level, so another PR would be better for that

**Review by matthieu-foucault — APPROVED** [review]

**Pull request merged**